tests/test_intake_dedup_and_url.py

```
@@ -0,0 +1,400 @@
+
+    def test_dedup_emits_audit_event(self, temp_vault):
+        proc = _make_processor(temp_vault, skip_deep_dive=True)
+        url = "https://example.com/audit-log-test"
+        proc.process_single_source(
+            _write_clipping(temp_vault, "first.md", url=url), dry_run=False,
+        )
+        proc.process_single_source(
+            _write_clipping(temp_vault, "second.md", url=url), dry_run=False,
+        )
+        log = (temp_vault / "60-Logs" / "pipeline.jsonl").read_text(encoding="utf-8")
+        events = [json.loads(l) for l in log.splitlines() if l.strip()]
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)